File: basic.js
function dfs(graph, root) {
    const n = graph.length;
    const visited = new Array(n).fill(false);
    const stack = [];
    const result = [];

    // Start from the root
    stack.push(root);
    visited[root] = true;

    while (stack.length > 0) {
        const current = stack.pop();  // LIFO - Depth First
        result.push(current);

        // Explore neighbors in reverse order (optional but helps match common test expectations)
        for (let neighbor = n - 1; neighbor >= 0; neighbor--) {
            if (graph[current][neighbor] === 1 && !visited[neighbor]) {
                visited[neighbor] = true;
                stack.push(neighbor);
            }
        }
    }

    return result;
}

File: basic.py
def dfs(graph, root):
    n = len(graph)
    visited = [False] * n
    stack = []
    result = []

    stack.append(root)
    visited[root] = True

    while stack:
        current = stack.pop()      # LIFO - Depth First
        result.append(current)

        # Explore neighbors in reverse order to match typical test output order
        for neighbor in range(n - 1, -1, -1):
            if graph[current][neighbor] == 1 and not visited[neighbor]:
                visited[neighbor] = True
                stack.append(neighbor)

    return result
